RELEASES.md

```diff
@@ -1,3 +1,105 @@
+Version 1.23.0 (2018-01-04)
+==========================
+
+Language
+--------
+- [Fixed displaying duplicate errors on some trait bounds.][45772]
+
+Compiler
+--------
+- [Enabled `TrapUnreachable` in LLVM which should prevent some cases of
+  undefined behaviour.][45920]
+- [rustc now suggests renaming import if names clash.][45660]
+- [Display errors/warnings correctly when there are zero-width or
+  wide characters.][45711]
+- [rustc now uses subtyping on the left hand side of binary operations.][45435]
+  Which should fix some confusing errors in some operations.
+- [Bumped the minimum LLVM to 3.9][45326]
+- [rustc now avoids unnecessary copies of arguments that are
+  simple bindings][45380] This should improve memory usage on average by 5-10%.
+- [Updated musl used to build musl rustc to 1.1.17][45393]
+
+Libraries
+---------
+- [impl `From<T>` for `Mutex<T>` and `RwLock<T>`][46082]
+- [Allow a trailling comma in `assert_eq/ne` macro][45887]
+- [Implement Hash for raw pointers to unsized types][45483]
+- [impl `From<*mut T>` for `AtomicPtr<T>`][45610]
+- [impl `From<{number_size}>` for `Atomic{number_size}`][45610] eg. `From<u8>`
+  for `AtomicU8`.
+- [Removed the `T: Sync` requirement for `RwLock<T>: Send`][45267]
+- [Removed `T: Sized` requirement for `<*const T>::as_ref`
+  and `<*const T>::as_mut`][44932]
+- [Optimized `Thread::{park, unpark}` implementation][45524]
+- [Improved `SliceExt::binary_search` performance.][45333]
+- [Optimized `Read::read_to_end`.][46050] This increase file read speed and for
+  small files reduce the number of syscalls.
+
+Stabilized APIs
+---------------
+
+Cargo
+-----
+- [Cargo now supports alternative registries][cargo/4506]
+- [Cargo now supports uninstallation of multiple packages][cargo/4561]
+  eg. `cargo uninstall foo bar` uninstalls `foo` and `bar`.
+- [Added unit test checking to `cargo check`][cargo/4592]
+- [Cargo now lets you install a specific version
+  using `cargo install --version`][cargo/4637]
+
+Misc
+----
+- [Releases now ship with the Cargo book documentation.][45692]
+- [rustdoc now prints rendering warnings on every run.][45324]
+- [Release tarballs now come with rustfmt][45903]
+
+Compatibility Notes
+-------------------
+- [`fmt::Arguments` can no longer be shared across threads.][45198]
+- [Changes have been made to type equality to make it more correct,
+  in rare cases this could break some code.][45853] [Tracking issue for
+  further information][45852]
+- [Changed how closures are stored, as a result of this change you can no longer
+  have a situation where a closure directly calls itself.][45879]
+- [`char::escape_debug` now uses Unicode 10 over 9.][45571]
+- [Upgraded Android SDK to 27, and NDK to r15c.][45580] This drops support for
+  Android 9, the minimum supported version is Android 14.
+- [Refactored type memory layouts and ABIs.][45225] This reduce code generated
+  for a lot of types.
```

Comment:
#45225 should be also part of the Compiler section. I don't know about "code generated", as the most visible effect of #45225 is the reduced type size (e.g. size of `Option<bool>` is now 1 byte instead of 2 bytes; or more realistically, `std::io::Result<()>` is now 16 bytes long instead of 24 bytes).